workers/unity/Packages/io.improbable.gdk.core/Subscriptions/CallbackManagers/CommandResponseCallbackManager.cs

```diff
@@ -23,20 +22,22 @@ public void InvokeCallbacks()
             for (var i = 0; i < responses.Count; ++i)
             {
                 ref readonly var response = ref responses[i];
-                callbacks.InvokeAll(response.RequestId, response);
-                callbacks.RemoveAllCallbacksForIndex(response.RequestId);
+                if (callbacks.TryGetValue(response.RequestId, out var callback))
+                {
+                    callbacks.Remove(response.RequestId);
+                    callback(response);
+                }
             }
         }
 
-        public ulong RegisterCallback(long requestId, Action<T> callback)
+        public void RegisterCallback(long requestId, Action<T> callback)
         {
-            callbacks.Add(requestId, nextCallbackId, callback);
-            return nextCallbackId++;
+            callbacks.Add(requestId, callback);
         }
 
-        public bool UnregisterCallback(ulong callbackKey)
+        public bool UnregisterCallback(ulong requestId)
```

Comment:
Why is this actually ulong instead of long?

Reply:
Turns out the command request id's are `long` instead of `ulong` which we use everywhere else. Should be fixed in a separate PR.